Firestore/core/src/firebase/firestore/model/document_key_set.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_MODEL_DOCUMENT_KEY_SET_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_MODEL_DOCUMENT_KEY_SET_H_
+
+#include <functional>
+#include <set>
+#include <string>
+
+#include "Firestore/core/src/firebase/firestore/model/document_key.h"
+
+namespace firebase {
+namespace firestore {
+namespace model {
+
+/** Convenience type for a set of keys, since they are so common. */
+typedef std::set<firebase::firestore::model::DocumentKey> DocumentKeySet;
```

Comment:
nit: Styleguide recommends 'using' over 'typedef' for this.
http://go/cpp-style#Aliases

`using DocumentKeySet = std::set<stuff>;`